### PR TITLE
[Python 2 to 3] Fix overwritten imports (small)

### PR DESCRIPTION
**Pull request by gpotter2**
This PR fixes a bug:
- new import system using `importlib`
- importing everything (even `sys`...) would make the err stream get lost

@guedou That's a small but quite import one :) Thanks !

### CODE REVIEW

**Comment by guedou** [comment]
Could you provide an example of the bug this PR is fixing ?

**Comment by gpotter2** [comment]
I can't right now, here how the bug is created:
- When switching to python 3, we need to replace `__builtin__` by `builtins`.
- When loading a module using `__import__`, it appears that are some compatibility problems with `builtins`, so the modules cannot be loaded properly. Anyway, it's cleaner to use [`importlib`](https://docs.python.org/3/library/importlib.html) on which the `import` keyword depends.
- Using `importlib` imports everything from the module. Which means that if we import the main modules like `sys` in a module, for a weird reason the imported `sys` overwrites the current one. That way the `err` stream is lost.

If we use `__import__` on Python 3 (i had tested this), then the bug still exist and is harder to fix.

As an example and proof, look at https://ci.appveyor.com/project/secdev/scapy/build/1.0.995, log of https://github.com/secdev/scapy/pull/630, where the bug exists. The error log just does not exist.

**Comment by gpotter2** [comment]
@guedou Could you please review this PR ? It's needed for the next Python 3 support ones

File: scapy/layers/all.py [hunk 1]
```diff
@@ -7,15 +7,22 @@
...
 from scapy.error import log_loading
-import logging
+import logging, importlib
+ignored = list(__builtin__.__dict__.keys()) + ["sys"]
 log = logging.getLogger("scapy.loading")
 
 __all__ = []
 
+
+def _validate_local(x):
```
Comment (guedou) [inline scapy/layers/all.py]: Could you add a doctring ?

File: scapy/main.py [hunk 2]
```diff
@@ -35,6 +37,9 @@ def _read_config_file(cf):
     except Exception,e:
         log_loading.exception("Error during evaluation of config file [%s]" % cf)
         
+def _validate_local(x):
```
Comment (guedou) [inline scapy/main.py]: same here.

File: scapy/layers/all.py (hunk 1 above)
Comment (gpotter2) [inline scapy/layers/all.py]: Done

Comment (guedou) [inline scapy/layers/all.py]: Why is this function also implemented in `main.py` ?

Comment (gpotter2) [inline scapy/layers/all.py]: it's used each time we want to import a module. it replaces the "name not beginning by _" test, that already existed in those modules

**Pull request merged**